### PR TITLE
Independent Neopixel strips option

### PR DESCRIPTION
**Pull request by Bacon-Fixation**
### Description
- Add (M150 Sn) an extension of the M150 command to use it as a blend color that is different from the first NeoPixel and Caselight.
- Add LCD Menu options for control of only the the separated neopixel2 pin 

### Benefits

- More control options on the NeoPixel pins

- Example: Say you're watching from a web cam you have the PrinterEvents on NeoPixel_pin after heat up they turn white and i want them to stay white through the print as normal but use M150 S1 to change the color of all of the LEDs on NeoPixel2_Seperate with varying intensities that are not set by the NeoPixel.
Cura adds labels to the Gcode like SHELL, INFILL,..... and doing a Search replace with M150 S1 and a color value
you can have this blending multi colored visual that let you kinda know at a glance at what phase of the layer the printer is at.

### Related Issues
- None that I know of

### CODE REVIEW

File: Marlin/Configuration.h [hunk 1]
```diff
@@ -946,8 +946,8 @@
 
 // Duet Smart Effector (for delta printers) - https://bit.ly/2ul5U7J
 // When the pin is defined you can use M672 to set/reset the probe sensivity.
-//#define DUET_SMART_EFFECTOR
-#if ENABLED(DUET_SMART_EFFECTOR)
+//#define SMART_EFFECTOR
+#if ENABLED(SMART_EFFECTOR)
```
Comment (Bacon-Fixation) [inline Marlin/Configuration.h]: this is misspelled when you compare it to the MarlinCore

**Comment by Bacon-Fixation** [comment]
Thank you for the help! I definitely need it <3

would like to still have a M151 command (or something different) if possible because wanted to use it as a blend color that is different from the first NeoPixel 

for example watching from a web cam you have the PrinterEvents on NeoPixel_pin after heat up they turn white and i want them to stay white through the print as normal but use M151 to change the color of all of the LEDs on NeoPixel2_Seperate with varying intensities that are not set by the NeoPixel.

Cura adds labels to the Gcode like SHELL, INFILL,..... and doing a Search replace with M151 and a color value 
you can have this blending multi colored visual that let you kinda know at a glance at what phase of the layer the printer is at.

would love to hear opinions

**Comment by thinkyhead** [comment]
`M150 Sn` is a reasonable substitute for `M151` and allows for any number of additional strips.

**Comment by thinkyhead** [comment]
My aim in further cleanup will be to get rid of the duplicated `Marlin_NeoPixel2` class and simply make `Marlin_NeoPixel` non-static or otherwise extend it so that it can address more than one Neopixel strip. The use of `static` classes in Marlin is to make the code smaller and faster. (Non-static methods have a hidden `this` argument.) If we make `Marlin_Neopixel` into a normal non-static class then we can have multiple unique instances which are managed by the `LEDLights` class. If we keep `Marlin_Neopixel` static but add the ability for it to represent multiple LED strips, then the class becomes slightly more complicated. The best-of-all solutions is to keep `Marlin_Neopixel` static when there's only one LED strip and non-static when there's more than one. That will keep the code as lean as it can be for the most common case.

**Comment by Bacon-Fixation** [comment]
Sounds Great. i see what the plan is now .... cool TY again

**Comment by Bacon-Fixation** [comment]
hey hey ran some real world testing with M150 Sn on my LPC1769 and here are my results 
NEOPIXEL_STARTUP_TEST bleeds into the second Channel
and NEOPIXEL2_STARTUP_TEST works fines
when we do a M150 S1 R127 the second channel works like it should turns on all leds in the second channel to red 
but if i run a M150 instead of  M150 S1  after the M150 S1 R127 half of the NeoPixels on the second channel will turn off and M150 as a hole will become unresponsive to new commands

but other commands like G28 and others still work when M150 is in an unresponsive state

**Comment by Bacon-Fixation** [comment]
whoops still new to github

**Comment by thinkyhead** [comment]
Yes, this PR is going to be requiring a lot of tearing down and rebuilding. There will be more changes applied over the coming days, and things are bound to get broken in the process, until they get fixed. Thanks for keeping me posted about the current state.

**Comment by SparkyDan555** [comment]
There was a recently merged PR that (if im understanding your query correctly) will do what you are trying to achieve. See #18992 . It allows NEOPIXEL2_PIN to work independently of NEOPIXEL1_PIN by giving the second strip LED indexes of their own. Simply set NEOPIXEL2_INSERIES to true and use M150. Please let me know if this is not what you are trying to achieve.

**Comment by Bacon-Fixation** [comment]
when i tested #18992 the printer event overrides pin2 when enabled and i cant set just some of the pixels of second pin it was all or nothing, 
but with M150 Sn we can have fully independent control of the pixels and the color that is not influenced by any other LED commands 

**Examples of wants:**
- M150 S1 R127 U127 B127 second pin White 
- M150 S1 I2 R for the Second Pixel of the second pin is red

**Comment by SparkyDan555** [comment]
Using the changes implemented in #18992, this is possible. Say for example you have 10 pixels on NEOPIXEL_PIN and 5 pixels on NEOPIXEL2_PIN and you wanted to set the 2nd pixel on NEOPIXEL_PIN to red. You would do this with M150 I1 R. To set the 2nd pixel on NEOPIXEL2_PIN to red you would send M150 I11 R. The only improvement I can see from your suggestions is the ability to set the entire NEOPIXEL2_PIN strip with one single command independently of the NEOPIXEL_PIN strip rather than issuing a separate M150 for every pixel on the second strip. Is this the intention?

**Comment by Bacon-Fixation** [comment]
TY for your reply <3
when I used #18992  there was no stopping printer events from influencing NEOPIXEL2_PIN, it's still tied to NEOPIXEL_PIN indirectly.
I want complete isolation from all other NEOPIXEL_PIN influences that is also separate from case lights

there isn't any LCD menu option to control ONLY NEOPIXEL2_PIN either  

I want to apologize if I didn't provide enough info :D its a learning experience for me

**Pull request merged**